mangdangroboticsclub/minipupper_ros
# Update control driver

Proposed Change(s)
------------------

This pull request proposes two main updates to support the Mini Pupper basic keyboard control without champ:

1.  `mini_pupper_control`: This package contains the `vel_to_servo_controller`, which receives the velocity topic `/cmd_vel` in ROS2 (with `/pose` to be added soon) and publishes the joints topic `/joint_command`.
    
2.  `mini_pupper_driver`: This package contains the `servo_interface` and `display_interface`. All hardware interface files should be configured in this package.
    

### Types of Change(s)


Testing and Verification
------------------------

To verify these changes, the following tests were performed:

1.  Launch `servo_interface` in `mini_pupper_driver`
2.  Launch `vel_to_servo_controller` in `mini_pupper_control`
3.  Run the Keyboard node

Test Configuration
------------------

**Mini Pupper Version**  
Mini Pupper

**Raspberry Pi OS + ROS version**  
Ubuntu 22.04, ROS 2 Humble

**PC OS + ROS version**  
Ubuntu 22.04, ROS 2 Humble

Checklist
---------

*   [x]  I have checked to ensure there aren't other open [Pull Requests](../pulls) for the exact change.

Other Comments
--------------

1.  Due to the missing MIT copyright in `StanfordQuadruped`, the MIT license was added to `StanfordQuadruped`. This change was made because the industrial-ci test checks for copyright information in files. Please share your thoughts and opinions on this matter.
2.  The submodule solution is not suitable as the import part in the Stanford Pupper package is complicated. Therefore, the Stanford-related components are placed in `mini_pupper_control/StanfordQuadruped`.
3.  The code style in `StanfordQuadruped` was changed to pass the flake8 test.
4.  Please add `pip install transforms3d` to the bash script. \[Important\]
